pybuildkite/client.py

```diff
@@ -1,27 +1,38 @@
+from __future__ import annotations
+
+from typing import Any, Dict, Iterator, TypeAlias, TYPE_CHECKING
+
 import requests
-from urllib.parse import urlparse
+
+from pybuildkite.response import Response
+
+if TYPE_CHECKING:
+    _QueryParams: TypeAlias = requests.sessions._Params
+
+_HeadersMapping: TypeAlias = Dict[str, Any]  # TODO: Bad Any.
+RequestResponse: TypeAlias = Response | bool | Any | Iterator[Any] | bytes
```

Comment:
This is probably the most important type and unfortunately because `request()` can return `response.json()` which is an `Any` (at least that's what [typeshed](https://github.com/python/typeshed/blob/3623f923a6647f92d1b276646f69c4927423c449/stubs/requests/requests/models.pyi#L139) specifies), this explicitly includes `or Any` making it basically useless :/. Not sure if it'll resolve the issue, but might be worthy of a refactor.